src/pymoca/backends/casadi/model.py

```diff
@@ -524,16 +524,42 @@ def simplify(self, options):
                         value = None
 
                     if variable is not None:
-                        del alg_states[variable.name()]
-
-                        variables.append(variable)
                         if eq.is_op(ca.OP_SUB):
-                            values.append(value)
+                            return variable, value
                         else:
-                            values.append(-value)
+                            return variable, -value
 
-                        # Skip this equation
-                        continue
+                if eq.n_dep() == 2 \
+                    and eq.is_op(ca.OP_ADD) \
+                    and eq.dep(0).is_op(ca.OP_IF_ELSE_ZERO) \
+                    and eq.dep(1).is_op(ca.OP_IF_ELSE_ZERO):
+
+                    cond_1 = eq.dep(0).dep(0)
+                    cond_2 = eq.dep(1).dep(0)
+                    nested_eq_1 = eq.dep(0).dep(1)
+                    nested_eq_2 = eq.dep(1).dep(1)
+
+                    variable_1, value_1 = extract_assignment(nested_eq_1)
+                    variable_2, value_2 = extract_assignment(nested_eq_2)
+
+                    if variable_1 is not None and variable_2 is not None and variable_1.name() == variable_2.name():
+                        return variable_1, ca.if_else(cond_1, value_1, 0, True) + ca.if_else(cond_2, value_2, 0, True)
```

Comment:
1) If this works for e.g. an `x + y`, will it also work for `z + x +y` (with all of them assigning to the same variable)
2) What if z is a different variable or cannot be resolved. There is no op/dep that matches just x + y, there is only one that does 'z + x` and one that does `(z + x) + y`

Reply:
See my response to the comment above.